### ISSUE TITLE
error while provisioning fresh box in MacOS

### ISSUE THREAD

**Issue by columbian-chris** [Varying-Vagrant-Vagrants/VVV#2530]
I was working on an issue (#2529) and thought I would try an attempt in a fresh VVV box.  I installed it but after I initially  provisioned it, it threw me errors:

```
    default: Reading package lists...
    default: E: The repository 'https://deb.nodesource.com/node_14.x focal Release' does not have a Release file.
    default: E: The repository 'https://nginx.org/packages/mainline/ubuntu focal Release' does not have a Release file.
    default:  ! The 'main' provisioner ran into problems, the full log is available at '/var/log/provisioners/2021.09.30_19-59-23/provisioner-main.log'. It completed in 30 seconds.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

I ssh-ed into the box and noticed that none of the default sites (like one.wordpress.test, etc.) were in the `/srv/www/` directory, so clearly the provisioning never reached that point of the process.

I looked at the full log file the error described and it didn't seem to provide anything additional.

**Comment by tomjn** [comment]
Sounds like you had the misfortune of provisioning while Node source had its pants down. It's like surprise guests turning up and ringing the doorbell while you're in the shower covered in soap. All I can suggest is that you wait a little while and try again.

**Comment by columbian-chris** [comment]
@tomjn Thanks for the response.  I tried again a second ago and still got the same error so I'll give it another try tomorrow.

**Comment by tomjn** [comment]
@columbian-chris we don't have any say or control over Nodesource, so unless you know of an alternative apt source there's nothing that can be done at our end

**Comment by tomjn** [comment]
perhaps destroy and recreate? That it's happening with Nginx too suggests either ISP intervention, terrible luck, or interference

**Comment by columbian-chris** [comment]
Yeah sorry, didn't work.  I'll give it another try tomorrow after more time has passed.  Do you know if Nodesource releases status notes or updates so we can know if they have problems going on?

**Comment by JenaMischk** [comment]
I'm experiencing the same issue on fresh installs, both on Windows and Pop!_OS 21.04. I've tried to set VVV up on Windows about 12h ago, and Pop!_OS just now. Tried to setup my local development in XAMPP as a temporary fix until the repositories come back up but that doesn't quite cut it, as the performance is miserable in comparison to VVV, and I am currently doing performance testing.

My setup fails at the exact same stage, complaining about:
    default: E: The repository 'https://deb.nodesource.com/node_14.x focal Release' does not have a Release file.
    default: E: The repository 'https://nginx.org/packages/mainline/ubuntu focal Release' does not have a Release file.

Destroying and recreating doesn't do the trick. This is an issue on their side, as I have been trying to setup for at least 12h, tried it multiple times at different time intervals and still no luck, their repositories haven't come online yet.
I'm in Portugal and my ISP is Vodafone, I assume @columbian-chris ISP is a different one, which rules out ISP intervention. Proxies or VPNs ain't it either.

@tomjn do you have any quick and dirty temporary fix that we can apply to circumvent these repositories? Maybe an alternative apt source and where to edit them?  Any suggestion would be appreciated.

**Comment by JenaMischk** [comment]
So, apparently some nodesource certificates have expiration dates set to the 30th of September, which broke quite some things.

Following https://github.com/nodesource/distributions/issues/1267 and https://github.com/nodesource/distributions/issues/1266 I did the following:
`vagrant ssh
sudo apt-get update
sudo apt-get install ca-certificates
sudo apt-get install libgnutls30`

After running these commands `vagrant up --provision` worked flawlessly for me, and I'm back at work.

**Comment by Mte90** [comment]
it is the day of issues of certificates in all t the world, all the systems and services needs updates.

**Comment by tomjn** [comment]
@JenaMischk I'm curious why `libgnutls30`?

**Comment by tomjn** [comment]
`ca-certificates` has been added to the main list of apt packages on the `develop` branch

**Comment by columbian-chris** [comment]
> Apparently DST Root CA X3 (older Let's Encrypt signed root certificate) expired yesterday, the 30th of September, which resulted in quite some broken things across the web.
> 
> For the particular case of nodesource, following [nodesource/distributions#1267](https://github.com/nodesource/distributions/issues/1267) and [nodesource/distributions#1266](https://github.com/nodesource/distributions/issues/1266) I took these steps: `vagrant ssh` `sudo apt-get update` `sudo apt-get install ca-certificates` `sudo apt-get install libgnutls30`
> 
> After running these commands `vagrant up --provision` worked as usual for me, and I'm back on track.

This also enabled me to successfully provision.

**Comment by shoreline-chrism** [comment]
I checked out the develop branch a few minutes before @columbian-chris commented and I still had the same issues with "focal Release doesn't have a release file.  Following @columbian-chris' instructions I was able to get my box spun up.  @tomjn Maybe more than ca-certificates is needed?  I think chris added the `libgnutls30` package as well

**Comment by shoreline-chrism** [comment]
> I still had the same issues with "focal Release doesn't have a release file.

I also destroyed the box and tried to run `vagrant up` on `develop` with same issues.

**Comment by tomjn** [comment]
Sounds like it is, I’m away from my machine but if somebody can do a pull
request to add it I can approve and merge, check the latest commit for
where to add it

**Comment by JenaMischk** [comment]
> @JenaMischk I'm curious why `libgnutls30`?

To be fair, it might very well run without it, although I can not test it right now.

On my first try to solve this issue I did install `libgnutls30` alone and the errors were not fixed. Only after installing `ca-certificates` was I able to run the provisioner. So, it is possible that my first attempt did nothing to it, and `ca-certificates` is enough in itself. But since so many people were referencing both `ca-certificates` and `libgnutls30` in their succesfull attempts to fix the issues at nodesource, I included both.

So, in my steps to fix the provisioner earlier today, I have tested provisioning with `libgnutls30` only (failed), then with both `libgnutls30` and `ca-certificates` (succeeded), but never with `ca-certificates` only. Will test and report as soon as possible.

**Comment by shoreline-chrism** [comment]
I forgot to link the issue but here's the PR: https://github.com/Varying-Vagrant-Vagrants/VVV/pull/2531

**Comment by tomjn** [comment]
I'm closing this as the fix is merged into `develop`, if you have this issue either switch to the `develop` branch or update to v3.8+ if it's been released
